File: internal/content/testdata/bench/java/gas-station.java
class Solution {
    public long can_complete_circuit(long[] gas, long[] cost) {
        long total = 0;
        for (int i = 0; i < gas.length; i++) total += gas[i] - cost[i];
        if (total < 0) return -1;
        long tank = 0;
        int start = 0;
        for (int i = 0; i < gas.length; i++) {
            tank += gas[i] - cost[i];
            if (tank < 0) {
                start = i + 1;
                tank = 0;
            }
        }
        return start;
    }
}

File: internal/content/testdata/bench/csharp/gas-station.cs
public class Solution {
    public long can_complete_circuit(long[] gas, long[] cost) {
        long totalGas = 0, totalCost = 0;
        foreach (long g in gas) totalGas += g;
        foreach (long c in cost) totalCost += c;
        if (totalGas < totalCost) return -1;
        long total = 0, start = 0;
        for (int i = 0; i < gas.Length; i++) {
            total += gas[i] - cost[i];
            if (total < 0) { start = i + 1; total = 0; }
        }
        return start;
    }
}
